fix DecompositionSeries link

bendichter opened the pull request:
Make DecompositionSeries.source_timeseries optional by setting quantity
to ?

## Motivation

fix #951

## Checklist

- [x] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [x] Have you ensured the PR description clearly describes problem and the solution?
- [x] Is your contribution compliant with our coding style ? This can be checked running `flake8` from the source directory.
- [x] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/NeurodataWithoutBorders/pynwb/pulls) for the same change?
- [x] Have you included the relevant issue number using `#XXX` notation where `XXX` is the issue number ?

bendichter committed:
fix DecompositionSeries link

Make DecompositionSeries.source_timeseries optional by setting quantity
to ?

bendichter committed:
Merge branch 'dev' into fix_decomp_series_link

oruebel commented:
I closed #952 as this PR changes this.

I created https://github.com/NeurodataWithoutBorders/nwb-schema/issues/264 to make sure we add the changes to the release notes when we update the schema repo.

oruebel reviewed (changes requested):
We should increase the version number in the namespace.yaml file to 2.0.3. 

Can should merge https://github.com/NeurodataWithoutBorders/pynwb/pull/953 first to avoid collisions.

oruebel commented:
How about adding the following warning to the constructor of ``DecompositionSeries``? I think it would be good to encourage folks to set ``source_timeseries`` if possible, even if it is not required.

```
+import warnings
+
 
 @register_class('AnnotationSeries', CORE_NAMESPACE)
 class AnnotationSeries(TimeSeries):
@@ -292,6 +294,9 @@ class DecompositionSeries(TimeSeries):
         metric, source_timeseries, bands = popargs('metric', 'source_timeseries', 'bands', kwargs)
         super(DecompositionSeries, self).__init__(**kwargs)
         self.source_timeseries = source_timeseries
+        if self.source_timeseries is None:
+            warnings.warn("Please set `source_timeseries` if available to document " +
+                          "where the DecompositionSeries is derived from. (Optional)")
         self.metric = metric
         if bands is None:
             bands = DynamicTable("bands", "data about the frequency bands that the signal was decomposed into")
```

bendichter committed:
add DecompSeries warning

Add warning to DecompositionSeries if source_timeseries is not present in constructor

bendichter committed:
Merge branch 'dev' into fix_decomp_series_link

oruebel reviewed (approved):
Looks good. We need to update the version number of the namespace, but I think #953 will take care of that. I think both this one and #953 should be good to go for merge. We just need to copy the schema then over the nwb-schema repo at some point and update the release notes. I already created an issue on the nwb-schema repo for this so we don't forget. Long story short, please merge both this one and #953.
